include/ginkgo/core/base/batch_lin_op.hpp

```diff
@@ -709,12 +702,19 @@ class EnableBatchScaling : public BatchScalable {
             GKO_NOT_SUPPORTED(this);
         }
 
+        left_scale_ = left_scale;
+        right_scale_ = right_scale;
+
         this->batch_scale_impl(left_scale, right_scale);
     }
 
 private:
     virtual void batch_scale_impl(const BatchLinOp* left_scale,
-                                  const BatchLinOp* right_scale) = 0;
+                                  const BatchLinOp* right_scale)
+    {}
```

Comment:
No, it's supposed to be optional. In fact, I can probably remove this virtual function soon, because after my latest changes none of the classes needs it. Earlier, batch matrices would actually call something here to do the scaling, whereas solvers would defer it to the apply step and do the scaling per-batch within the solve kernel.